main.cpp

```diff
@@ -19,13 +19,25 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
+    CastCallBack(Rewriter& rewriter):_rewriter(rewriter) {
         // Your code goes here
     };
 
     void run(const MatchFinder::MatchResult &Result) override {
         // Your code goes here
+	auto cstmtptr = Result.Nodes.getNodeAs<clang::CStyleCastExpr>("cast");
+	auto l_brace = cstmtptr->getLParenLoc();
+	auto r_brace = cstmtptr->getRParenLoc();

```

Comment:
Please describe the difference between `getSubExprAsWritten()->getBeginLoc()` and `getRParenLoc().getLocWithOffset(1)`.

Reply:
Using our test example, we can see what is the difference between getSubExprAsWritten()->getBeginLoc() and getRParenLoc().getLocWithOffset(1). 
getSubExprAsWritten()->getBeginLoc() points to the beginning of the variable fooooo.
Meanwhile, getRParenLoc().getLocWithOffset(1) points to the character right after ")".